### ISSUE TITLE
Erroneous suggestion IDE0067 for disposable object used in 'await using (...) { }'

### ISSUE THREAD

**Issue by Joe4evr** [dotnet/roslyn#38944]
**Version Used**:
Version 16.4.0 Preview 1.0
VisualStudio.16.Preview/16.4.0-pre.1.0+29319.158

**Steps to Reproduce**:
Paste the following code into a project:
```cs
using System;
using System.IO;
using System.Threading.Tasks;

static class DisposingAnalysis
{
    static void ImplicitSync(ReadOnlySpan<byte> bytes)
    {
        //No suggestions, expected
        using var stream = new MemoryStream();
        stream.Write(bytes);
    }

    static async Task ImplicitAsync(ReadOnlyMemory<byte> bytes)
    {
        //No suggestions, expected
        await using var stream = new MemoryStream();
        await stream.WriteAsync(bytes);
    }

    static void ExplicitSync(ReadOnlySpan<byte> bytes)
    {
        //IDE0063: 'using' statement can be simplified, expected
        using (var stream = new MemoryStream())
        {
            stream.Write(bytes);
        }
    }

    static async Task ExplicitAsync(ReadOnlyMemory<byte> bytes)
    {
        //IDE0063 *AND* IDE0067: Disposable object created by 'new MemoryStream()' is never disposed
        await using (var stream = new MemoryStream())
        {
            await stream.WriteAsync(bytes);
        }
    }
}
```

**Expected Behavior**:
Only IDE0063 is reported for both of the explicit syntax methods to point out the new Implictly Scoped `using` Statement syntax.

**Actual Behavior**:
The combination of the older, explicitly-scoped syntax and async disposable makes the compiler/IDE think the object will not get disposed and reports IDE0067 in addition to the expected behavior.
